test/examples/uct_hello_world.c

```diff
@@ -1,17 +1,635 @@
+            } else {
+                fprintf(stderr, "Unknown option character `\\x%x'.\n", optopt);
+            }
+        case 'h':
+        default:
+            return print_err_usage();
+        }
+    }
+    fprintf(stderr, "INFO: UCT_HELLO_WORLD AM function = %s server = %s port = %d\n",
+            func_am_t_str(args->func_am_type), args->server_name,
+            args->server_port);
+
+    for (index = optind; index < argc; index++) {
+        fprintf(stderr, "WARNING: Non-option argument %s\n", argv[index]);
+    }
+
+    if (args->dev_name == NULL) {
+        fprintf(stderr, "WARNING: device is not set\n");
+        return print_err_usage();
+    }
+
+    if (args->tl_name == NULL) {
+        fprintf(stderr, "WARNING: transport layer is not set\n");
+        return print_err_usage();
+    }
+
+    return UCS_OK;
+}
+
+/* The caller is responsible to free *rbuf */
+int sendrecv(int sock, const void *sbuf, size_t slen, void **rbuf)
+{
+    int ret = 0;
+    size_t rlen = 0;
+    *rbuf = NULL;
+
+    ret = send(sock, &slen, sizeof(slen), 0);
+    if ((ret < 0) || (ret != sizeof(slen))) {
+        fprintf(stderr, "failed to send buffer length\n");
+        return -1;
+    }
+
+    ret = send(sock, sbuf, slen, 0);
+    if ((ret < 0) || (ret != slen)) {
+        fprintf(stderr, "failed to send buffer\n");
+        return -1;
+    }
+
+    ret = recv(sock, &rlen, sizeof(rlen), 0);
+    if (ret < 0) {
+        fprintf(stderr, "failed to receive device address length\n");
+        return -1;
+    }
+
+    *rbuf = calloc(1, rlen);
+    if (!*rbuf) {
+        fprintf(stderr, "failed to allocate receive buffer\n");
+        return -1;
+    }
+
+    ret = recv(sock, *rbuf, rlen, 0);
+    if (ret < 0) {
+        fprintf(stderr, "failed to receive device address\n");
+        return -1;
+    }
+
+    return 0;
+}
+
+int main(int argc, char **argv)
+{
+    uct_device_addr_t   *own_dev;
+    uct_device_addr_t   *peer_dev = NULL;
+    uct_iface_addr_t    *own_iface;
+    uct_iface_addr_t    *peer_iface = NULL;
+    uct_ep_addr_t       *own_ep;
+    uct_ep_addr_t       *peer_ep = NULL;
+    ucs_status_t        status = UCS_OK; /* status codes for UCS */
+    uct_ep_h            ep;     /* Remote endpoint */
+    ucs_async_context_t async;  /* Async event context manages times and fd notifications */
+    cmd_args_t          cmd_args;
+
+    struct iface_info   if_info;
+    uint8_t             id = 0;
+    int                 oob_sock = -1;  /* OOB connection socket */
+
+    /* Parse the command line */
+    if (parse_cmd(argc, argv, &cmd_args)) {
+        status = UCS_ERR_INVALID_PARAM;
+        goto out;
+    }
+
+    /* Initialize context */
+    status = ucs_async_context_init(&async, UCS_ASYNC_MODE_THREAD);
+    CHKERR_JUMP(UCS_OK != status, "init async context", out);
+
+    /* Create a worker object */
+    status = uct_worker_create(&async, UCS_THREAD_MODE_SINGLE, &if_info.worker);
```

Comment:
If we use NULL as the first parameter we can avoid using async.h and avoid related issues.